src/compiler/scala/tools/nsc/classpath/AggregateClassPath.scala

```diff
@@ -117,20 +116,19 @@ case class AggregateClassPath(aggregates: Seq[ClassPath]) extends ClassPath {
         count += 1
       }
     }
-    mergedEntries.toIndexedSeq
+    if (mergedEntries isEmpty) Nil else mergedEntries.toIndexedSeq
   }
 
   private def getDistinctEntries[EntryType <: ClassRepresentation](getEntries: ClassPath => Seq[EntryType]): Seq[EntryType] = {
     val seenNames = collection.mutable.HashSet[String]()
     val entriesBuffer = new ArrayBuffer[EntryType](1024)
     for {
       cp <- aggregates
-      entry <- getEntries(cp) if !seenNames.contains(entry.name)
+      entry <- getEntries(cp)
     } {
-      entriesBuffer += entry
-      seenNames += entry.name
+      if (seenNames.add(entry.name)) entriesBuffer += entry
     }
-    entriesBuffer.toIndexedSeq
+    if (entriesBuffer isEmpty) Nil else entriesBuffer.toIndexedSeq
   }
 }
```

Comment:
For the next round of changes, `ClassPath#{list,classes, etc}` would compose better (from a performance perspective) if they accepted a buffer as an argument rather than returning collections.

Reply:
this PR was just the changes that dont affect the behaviour, structure etc